### ISSUE TITLE
Reflected time on some segments of SensorGlucoseReadingsEvent

### ISSUE THREAD

**Issue by mortlind** [pazaan/decoding-contour-next-link#31]
This issue is probably a remnant of issue #28, which indicates that that issue was only partially solved.

In the graph included in the screenshot below, observe the sensor glucose values from time -40 min. to 0 min. There can be observed two sequences of five and three points, which are separated by apparant discontinuities. If those two sequences are reflected in time around their respective centres, the whole graph becomes continuous, up to discretization. 

Especially note the earliest (~7 mM) and latest (~8 mM) values of the latest segment, and note the current SG value (7.10 mM) at the top if the image. The current value should correspond to the latest value in the graph, but instead it corresponds to the first value in the latest segment of the graph. If that segment was reflected in time around it's centre, there would be a match.

It is further clear, that not all segments, presumably each contained in a `SensorGlucoseReadingsEvent` object, should be reflected in time. The main trajectory leading up to t = -40 min is a segment which should not be reflected. When looking into the data structures, I find no obvious data element to tell, whether a segment should be reflected or not. 

![image](https://user-images.githubusercontent.com/5364433/34218502-a0ed4810-e5ae-11e7-8084-5655d2e8c105.png)

**Comment by mortlind** [comment]
The following patch solves the issue for me.

```diff
diff --git a/pump_history_parser.py b/pump_history_parser.py                                                                         
index a94d397..e371faf 100644                                                                                                        
--- a/pump_history_parser.py                                                                                                         
+++ b/pump_history_parser.py                                                                                                         
@@ -397,7 +397,7 @@ class SensorGlucoseReadingsEvent(NGPHistoryEvent):                                                               
                                                                                                                                     
     def allNestedEvents(self):                                                                                                      
         pos = 15                                                                                                                    
-        for i in range(self.numberOfReadings - 1, -1, -1):                                                                          
+        for i in range(self.numberOfReadings):^M                                                                                    
             #const timestamp = new NGPUtil.NGPTimestamp(this.timestamp.rtc - (i * this.minutesBetweenReadings * 60), this.timestamp.offset);                                                                                                                             
             timestamp = self.timestamp - timedelta(minutes = i * self.minutesBetweenReadings)                                       
             payloadDecoded = struct.unpack( '>BBHBhBB', self.eventData[pos:pos + 9] )       
```

**Comment by mortlind** [comment]
In addition, for using `read_minimed_next24.py` as a module, I would like to suggest the following patch:

```diff
diff --git a/read_minimed_next24.py b/read_minimed_next24.py                                                                         
index 380126a..ab6a19a 100644                                                                                                        
--- a/read_minimed_next24.py                                                                                                         
+++ b/read_minimed_next24.py                                                                                                         
@@ -1,8 +1,10 @@                                                                                                                     
 #!/usr/bin/env python                                                                                                               
                                                                                                                                     
 import logging                                                                                                                      
-# logging.basicConfig has to be before astm import, otherwise logs don't appear                                                     
-logging.basicConfig(format='%(asctime)s %(levelname)s [%(name)s] %(message)s', level=logging.WARNING)                               
+                                                                                                                                    
+if __name__ == '__main__':                                                                                                          
+    # logging.basicConfig has to be before astm import, otherwise logs don't appear                                                 
+    logging.basicConfig(format='%(asctime)s %(levelname)s [%(name)s] %(message)s', level=logging.WARNING)                           
 # a nasty workaround on missing hidapi.dll on my windows (allows testing from saved files, but not download of pump)                
 try:                                                                                                                                
     import hid # pip install hidapi - Platform independant
```

**Comment by paulokow** [comment]
Fix merged to master with 18b1bfd
